### PR TITLE
Add ::retrieveExtensionMetadataByName; update DMRIInstall helper

### PR DESCRIPTION
**Pull request by ihnorton**
- 2e2fcef adds a utility function `qSlicerExtensionsManagerModel::retrieveExtensionMetadataByName`
- dcc57af uses this utility to query metadata, by name, from Python and then run an installer task directly (plus other misc updates).

### CODE REVIEW

File: Modules/Scripted/DMRIInstall/DMRIInstall.py [hunk 1]
```diff
@@ -58,20 +85,43 @@ def setup(self):
...
   def onApply(self):
-    slicer.app.openExtensionsManagerDialog()
+    emm = slicer.app.extensionsManagerModel()
+
+    if emm.isExtensionInstalled("SlicerDMRI"):
+      self.textBox.setHtml("<h4>SlicerDMRI is already installed.<h4>")
+      self.applyButton.enabled = False
+      return
+
+    md = emm.retrieveExtensionMetadataByName("SlicerDMRIFoo")
```
Comment (jcfr) [inline Modules/Scripted/DMRIInstall/DMRIInstall.py]: Is `SlicerDMRIFoo` the expected name ?

File: Modules/Scripted/DMRIInstall/DMRIInstall.py [hunk 2]
```diff
@@ -58,20 +85,43 @@ def setup(self):
...
+
+    if md == {} or not md.has_key('extension_id'):
```
Comment (jcfr) [inline Modules/Scripted/DMRIInstall/DMRIInstall.py]: `if not md or not md.has_key('extension_id'):`

File: Modules/Scripted/DMRIInstall/DMRIInstall.py (hunk 1 above)
Comment (ihnorton) [reply Modules/Scripted/DMRIInstall/DMRIInstall.py]: Hah, oops. I forgot to tack out the failure-mode test. Thanks!

**Comment by ihnorton** [comment]
Updated

**Comment by lassoan** [comment]
Thank you, looks good to me!

**Comment by jcfr** [comment]
Integrated in [r26820](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26820) and [r26821](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26821)

**Comment by ihnorton** [comment]
I noticed that `qRestAPI::Parameters` is just a `QMap<string,string>` typedef, so this could be reworked for more general use as in the following commits:

https://github.com/Slicer/Slicer/compare/master...ihnorton:update_extsmgr_and_dmriinstallutil

Then the input can be auto-converted from a Python dict, and we could use it from Python to query arbitrary metadata combinations instead of only extension-by-name (for example to simplify the Midas query implementation in [ExtensionStats](https://github.com/fbudin69500/SlicerDeveloperToolsForExtensions/blob/master/ExtensionStats/ExtensionStats.py)).

One question though, should we add an API to return an array if there are multiple results? Right now there are internal checks whether more than one result was returned from the query, in which case the function just returns an empty dict.

**Pull request closed without merging**